utils/PathSanitizingFileCheck

```diff
@@ -72,7 +74,16 @@ constants.""")
         # We are replacing the Unix path separators in the paths passed as
         # arguments with a broader pattern to also allow forward slashes and
         # double escaped slashes in the result that we are checking. Sigh.
-        stdin = re.sub(re.sub(r'/', slashes_re, pattern), replacement, stdin)
+        pattern = re.sub(r'/', slashes_re, pattern)
+        stdin = re.sub(pattern, replacement, stdin)
+
+        if args.enable_windows_compatibility:
+            # For making easier to write the tests, we also look in the
+            # possible subpath attached to the replacement.
+            extended_replacement = replacement + r'[^\s\'"]+'
```

Comment:
Why do we need to deal with the quoting now when we didn't previously?

Reply:
We are not doing something that we were doing before.

Before, the only thing that we were doing was matching the path found in the result and transforming it into the tokens `SOURCE_DIR` or `BUILD_DIR`. If the output had `c:\s\b\swift` it was replaced by `BUILD_DIR`.

This new piece of code does something similar, but not the same: it looks for things that starts with `BUILD_DIR` (for example) and keeps looking until it finds an space, or some quote. Then it replaces all the `\` in there for the Unix `/`, so they match the tests. The space and the quotes are the typical stop characters for paths, and the ones that cover all the cases I could find in the existing tests.

In summary: first replacement replaces `c:\s\b\swift\foo\bar\baz` into `BUILD_DIR\foo\bar\baz`, and the second takes that result and replaces it for `BUILD_DIR/foo/bar/baz`. Technically you can try to do both at the same time, but I wanted to not modify the Unix path, if possible. This is clearly under the Windows compatibility mode.